components/script/dom/globalscope.rs

```diff
@@ -812,6 +813,13 @@ impl GlobalScope {
     pub fn get_user_agent(&self) -> Cow<'static, str> {
         self.user_agent.clone()
     }
+
+    pub fn get_csp_list(&self) -> Option<CspList> {
+        if let Some(window) = self.downcast::<Window>() {
+            return window.Document().get_csp_list().map(|c| c.clone());
```

Comment:
Why clone? And why don't other kinds of globals have any CSP list?

Reply:
> Why clone?

The `Document` function's return value is too short to return a reference to the `CspList`. I tried; borrowck throws an error.

> And why don't other kinds of globals have any CSP list?

Other kinds of globals do have a CSP list, as described at <https://www.w3.org/TR/CSP/#get-csp-of-object>. I just haven't implemented it yet.